components/core/src/clp/clo/CommandLineArguments.cpp

```diff
@@ -225,197 +103,510 @@ CommandLineArguments::parse_arguments(int argc, char const* argv[]) {
-            cerr << "OUTPUT_HANDLER is one of:" << endl;
-            cerr << "  " << static_cast<char const*>(cNetworkOutputHandlerName)
-                 << " - Output to a network destination" << endl;
-            cerr << "  " << static_cast<char const*>(cResultsCacheOutputHandlerName)
-                 << " - Output to the results cache" << endl;
-            cerr << "  " << static_cast<char const*>(cReducerOutputHandlerName)
-                 << " - Output to the reducer" << endl;
-            cerr << endl;
-
-            cerr << "Examples:" << endl;
-            cerr << R"(  # Search ARCHIVE_PATH for " ERROR " and send results to )"
-                    "a network destination"
-                 << endl;
-            cerr << "  " << get_program_name() << R"( ARCHIVE_PATH " ERROR ")"
-                 << " " << static_cast<char const*>(cNetworkOutputHandlerName)
-                 << " --host localhost --port 18000" << endl;
-            cerr << endl;
-
-            cerr << R"(  # Search ARCHIVE_PATH for " ERROR " and output the results )"
-                    "by performing a count aggregation"
-                 << endl;
-            cerr << "  " << get_program_name() << R"( ARCHIVE_PATH " ERROR ")"
-                 << " " << static_cast<char const*>(cReducerOutputHandlerName) << " --count"
-                 << " --host localhost --port 14009 --job-id 1" << endl;
-            cerr << endl;
-
-            cerr << R"(  # Search ARCHIVE_PATH for " ERROR " and send results to)"
-                    R"( mongodb://127.0.0.1:27017/test "result" collection )"
-                 << endl;
-            cerr << "  " << get_program_name() << R"( ARCHIVE_PATH " ERROR ")"
-                 << " " << static_cast<char const*>(cResultsCacheOutputHandlerName)
-                 << R"( --uri mongodb://127.0.0.1:27017/test --collection result)" << endl;
-            cerr << endl;
-
-            cerr << "Options can be specified on the command line or through a configuration file."
-                 << endl;
-            cerr << visible_options << endl;
-            return ParsingResult::InfoCommand;
-        }
-
         // Handle --version
         if (parsed_command_line_options.count("version")) {
             cerr << static_cast<char const*>(cVersion) << endl;
             return ParsingResult::InfoCommand;
         }
 
-        // Validate archive path was specified
-        if (m_archive_path.empty()) {
-            throw invalid_argument("ARCHIVE_PATH not specified or empty.");
-        }
+        // Validate command
+        if (parsed_command_line_options.count("command") == 0) {
+            // Handle --help
+            if (parsed_command_line_options.count("help")) {
+                if (argc > 2) {
+                    SPDLOG_WARN("Ignoring all options besides --help.");
+                }
+
+                print_basic_usage();
+                cerr << "COMMAND is one of:" << endl;
+                cerr << "  s - search" << endl;
```

Comment:
Wherever we have `s` or `i`, let's use enum_to_underlying_type with the command.